src/partialcor.jl

```diff
@@ -0,0 +1,62 @@
+# Partial correlation
+
+export PartialCorTest
+
+"""
+    PartialCorTest(x, y, Z)
+
+Perform a t-test for the hypothesis that ``\\text{Cor}(x,y|Z=z) = 0``, i.e. the partial
+correlation of vectors `x` and `y` given the matrix `Z` is zero.
+
+Implements `pvalue` and `confint`. See also [`partialcor`](@ref).
+"""
+struct PartialCorTest{T<:Real} <: HypothesisTest
+    r::T
+    n::Int
+    k::Int
+    t::T
+
+    # Error checking is done in `partialcor`
+    function PartialCorTest(x::AbstractVector, y::AbstractVector, Z::AbstractMatrix)
+        r = partialcor(x, y, Z)
+        n, k = size(Z)
+        t = r * sqrt((n - 2 - k) / (1 - r^2))
+        return new{typeof(r)}(r, n, k, t)
+    end
+
+    function PartialCorTest(x::AbstractVector, y::AbstractVector, z::AbstractVector)
+        r = partialcor(x, y, z)
+        n = length(z)
+        t = r * sqrt((n - 3) / (1 - r^2))
+        return new{typeof(r)}(r, n, 1, t)
+    end
+end
+
+testname(::PartialCorTest) = "Test for partial correlation"
+population_param_of_interest(p::PartialCorTest) = ("Partial correlation", zero(p.r), p.r)
+
+StatsBase.nobs(p::PartialCorTest) = p.n
+StatsBase.dof(p::PartialCorTest) = p.n - 2 - p.k
+
+function StatsBase.confint(test::PartialCorTest{T}, alpha::Float64=0.05) where T
+    df = dof(test)
+    df > 1 || return (-one(T), one(T))  # Otherwise we can get NaNs
+    q = quantile(TDist(df), 1 - alpha / 2)
```

Comment:
```suggestion
    q = quantile(Normal(), 1 - alpha / 2)
```
Let's go back to Normal quantiles for the confidence interval as the more common choice.